packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -598,28 +598,82 @@ describe("SharedTree", () => {
 
 			const log: string[] = [];
 			const unsubscribe = root[on]("changing", () => log.push("change"));
+			const unsubscribeSubtree = root[on]("subtreeChanging", () => log.push("subtree"));
 			const unsubscribeAfter = tree1.events.on("afterBatch", () => log.push("after"));
 			log.push("editStart");
 			root[valueSymbol] = 5;
 			log.push("editStart");
 			root[valueSymbol] = 6;
 			log.push("unsubscribe");
 			unsubscribe();
+			unsubscribeSubtree();
 			unsubscribeAfter();
 			log.push("editStart");
 			root[valueSymbol] = 7;
 
 			assert.deepEqual(log, [
 				"editStart",
+				"subtree",
 				"change",
 				"after",
 				"editStart",
+				"subtree",
 				"change",
 				"after",
 				"unsubscribe",
 				"editStart",
 			]);
 		});
+
+		it("propagates path and value args for local and subtree changes", async () => {
```

Comment:
While its good to have some integration testing of the events, I think it would be preferred to do the detailed correctness testing in more targeted unit tests.

It should be possible to setup a build a forest with, and get an editable tree for that without involving the shared tree class or fluid at all. src/test/feature-libraries/editable-tree/editableTree.spec.ts has some examples of this.

I think manually triggering the events from the forest's anchorset (even if that requires casting its events object to an emitter) might be a cleaner way to unit test this (you don't actually have to edit the tree that way, and can just check the events subscribe and unsubscribe properly).

With just these two tests, I'm fine leaving it here as if you want, but if we are going to get significantly more of them in the future, I'd prefer more targeted unit tests.

Reply:
Added `editable-tree/editableTree.events.spec.ts` dedicated tests, thus resolving this conversation